server.cjs

```diff
@@ -0,0 +1,206 @@
+require('dotenv').config();
+const express = require('express');
+const cors = require('cors');
+const app = express();
+const port = process.env.PORT;
+const supabaseUrl = process.env.VITE_URL;
+const supabaseKey = process.env.VITE_KEY;
+
+const { createClient } = require('@supabase/supabase-js');
+
+// Inicjacja Supabase klienta
+const supabase = createClient(
+    supabaseUrl,
+    supabaseKey
+);
+
+app.use(express.json());
+
+app.use(cors({
+  origin: 'http://localhost:8080', // Adres URL frontendu
+  credentials: true,
+}));
+
+// Żądania, które komunikują się z bazą Supabase
+app.get('/expenses/:userId', async (req, res) => {
+  try {
+    const userId = req.params.userId;
+    const { data, error } = await supabase.from('expenses').select('*').eq('id', userId);
+    if (error) throw error;
+    res.json(data);    
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: 'Internal Server Error' });
+  }
+});
+
+app.get('/income/:userId', async (req, res) => {
+  try {
+    const userId = req.params.userId;
+    const { data, error } = await supabase.from('income').select('*').eq('id', userId);
+    if (error) throw error;
+    res.json(data);    
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: 'Internal Server Error' });
+  }
+});
+
+app.get('/getMonths', async (req, res) => {
+  try {
+    const userId = req.params.userId;
+    const { data, error } = await supabase.from('income')
+    .select('incomeId, monthName, year')
+    if (error) throw error;
+    res.json(data);    
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: 'Internal Server Error' });
+  }
+});
+
+app.post('/addExpense', async (req, res) => {
+  try {
+    const expense = req.body.values.expense;
+    const price = req.body.values.price;
+    const userId = req.body.userId;
+    const idFormat = req.body.idFormat;
+    const productLabel = req.body.productLabel;
+
+    const { data, error } = await supabase
+    .from('expenses')
+    .insert({ id: userId, productCategory: expense, productPrice: price, created_at: idFormat, productLabel: productLabel })
+    .select()
+    if (error) throw error;
+    res.json(data);
+
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: 'Internal Server Error' });
+  }
+});
+
+app.post('/updateExpense', async (req, res) => {
+  try {
+    const expense = req.body.values.expense;
+    const price = req.body.values.price;
+    const id = req.body.id;
+
+    const { data, error } = await supabase
+    .from('expenses')
+    .update({ productCategory: expense, productPrice: price })
+    .eq('productLabel', id)
+    .select()
+    if (error) throw error;
```

Comment:
It would be good to distinguish between different errors and then return different HTTP responses. For example, if a record was not found, you can raise 404